pkg/apis/core/types.go

```diff
@@ -335,6 +335,14 @@ type PersistentVolumeSpec struct {
 	// This field influences the scheduling of pods that use this volume.
 	// +optional
 	NodeAffinity *VolumeNodeAffinity
+	// Name of VolumeAttributesClass to which this persistent volume belongs. Empty value
```

Comment:
I think similar to what we've done with StorageClass, we will want to distinguish between empty and nil values, so that users can indicate they don't want any VAC. Right now nil means that the default will be applied (and that should also be documented here).

Reply:
Yes that should be allowed

Reply:
@msau42 

For static provisioning, if a pvc has a vac and there're 2 pv, one has this vac and another doesn't has vac, which pv will be bound by pvc? In other words, Does vacs take affect on the existing bound logic? BTW, If a user create a pv with a vac specified, but this volume actually doesn't has this attributes defined in the vac. What happened? If the pv doesn't belongs to csi volume when a user create it, what happened? In which case users want to changes the vac of a pv from an empt
y value to other in the static provisioning case. If we want to allows users to indicate they don't want any VAC on a pv, Is it enough t
o apply only this rule on the pvc definition and disallow empty value on the pv definition.

Reply:
These are good questions. @sunnylovestiramisu do you want to think about these corner cases?

Reply:
Oh, I just realized this is on the PV object. I meant for my original comment to be referring to PVC object.

For PV object, nil is fine to mean does not belong to any VAC. This is inconsistent with PV.Spec.StorageClassName which is a non-pointer field, but that was before we added the guidance that all new fields should be pointers.

Reply:
PersistentVolumeClaimSpec.VolumeAttributesClassName:
* nil has two meanings: 1. if the PVC is bound, no VAC, similar to empty case. 2. if the PVC is unbound, the user wants to set with the default VAC
* empty means no VAC

Reply:
For static provisioning, if a pvc has a vac and there're 2 pv, one has this vac and another doesn't has vac, which pv will be bound by pvc?

Answer: vac in the pvc and pv needs to match.


If a user create a pv with a vac specified, but this volume actually doesn't has this attributes defined in the vac. What happened?

Answer: Only admin creates the VAC, so if they made a mistake, we still bind.


In which case users want to changes the vac of a pv from an empty value to other in the static provisioning case.
Answer: we should allow this behavior.

If the pv doesn't belongs to csi volume when a user create it, what happened? 
Answer: This is a csi feature so we should not bind this PV, we can add validation at PV controller that you can only set VAC for a csi volume.

Reply:
Admin creates the PV object. So we put more trust that they will do the right thing.

Also I think the pv csi validation could also be done in api validation.

Reply:
Who is responsible for setting the CurrentVolumeAttributesClassName field in the static provision, pv controller, csi driver or pvc creator?

Reply:
> Admin creates the PV object. So we put more trust that they will do the right thing.
> 
> Also I think the pv csi validation could also be done in api validation.

@msau42 It's added in https://github.com/kubernetes/kubernetes/pull/121104/commits/0c9fae3e297ebec32aab5f86c342c045ba08c39b, please review again.